src/go/k8s/internal/client/users/password.go

```diff
@@ -40,14 +40,18 @@ func (p *passwordGenerator) Generate() (string, error) {
 	if err != nil {
 		return "", err
 	}
-	password.WriteByte(p.firstCharacters[index])
+	if err := password.WriteByte(p.firstCharacters[index]); err != nil {
```

Comment:
Sorry, I was not very clear at all here. I wasn't asking for a error check. I don't think `strings.Builder` can return errors in these cases IIRC, kinda like how hashes implement `io.Writer` but never return errors. I was musing about golangci-lint either missing something or having an oddly helpful special case in place.